src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -743,6 +757,10 @@ def cleanup_nodes(self) -> bool:
                     # only add nodes that are not already set to reschedule
                     to_reimage.append(node)
 
+        dead_nodes = Node.get_dead_nodes(self.scaleset_id, NODE_EXPIRATION_TIME)
+        for node in dead_nodes:
```

Comment:
Adding `node.set_halt()` would be good to add here, just in case the node comes back alive while we're trying to reset it.